### PR TITLE
Allow pretty ser to work with implicit-some extension

### PR DESCRIPTION
**Pull request by cmaher**
Additionally configure newline for tests to work on windows.

I was converting my yaml files to ron and needed this, so I figured I'd submit a PR.  Really enjoying RON as a data language, btw!

### CODE REVIEW

File: src/ser/mod.rs [hunk 1]
```diff
@@ -112,6 +96,8 @@ pub struct PrettyConfig {
     /// Enumerate array items in comments
     #[serde(default = "default_enumerate_arrays")]
     pub enumerate_arrays: bool,
+    /// Enable implicit_some extension
+    pub implicit_some: bool,
```
Comment (kvark) [inline src/ser/mod.rs]: We would potentially want other features there as well. Let's make it a bitflags field to account for future needs?
We could move the `struct_names` in these flags too, sometimes later.

Comment (cmaher) [reply src/ser/mod.rs]: Sure. I'm going to be busy for the next week, but I'll try to get to it after that.

**Comment by cmaher** [comment]
Sorry for the delay-- I factored out the Extensions from `parse`, and adding it as a field on `PrettyConfig` (replacing the `implicit_some` in the initial commit). It's a bit awkward to not have it handle the struct_names, but I didn't see a clean way of handling that without breaking the existing API, which I didn't want to do here.

File: src/ser/mod.rs [hunk 2]
```diff
@@ -207,6 +208,7 @@ impl Default for PrettyConfig {
             indentor: default_indentor(),
             separate_tuple_members: default_separate_tuple_members(),
             enumerate_arrays: default_enumerate_arrays(),
+            extensions: default_extensions(),
```
Comment (kvark) [inline src/ser/mod.rs]: should probably just use `Extensions::default()` here

Comment (cmaher) [reply src/ser/mod.rs]: ah, good point. will fix that right up.

**Pull request merged**